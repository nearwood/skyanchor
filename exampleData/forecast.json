{
    "@context": [
        "https://geojson.org/geojson-ld/geojson-context.jsonld",
        {
            "@version": "1.1",
            "wx": "https://api.weather.gov/ontology#",
            "geo": "http://www.opengis.net/ont/geosparql#",
            "unit": "http://codes.wmo.int/common/unit/",
            "@vocab": "https://api.weather.gov/ontology#"
        }
    ],
    "type": "Feature",
    "geometry": {
        "type": "Polygon",
        "coordinates": [
            [
                [
                    -91.163689899999994,
                    30.417998600000001
                ],
                [
                    -91.164435799999993,
                    30.395269200000001
                ],
                [
                    -91.138086699999988,
                    30.394623800000002
                ],
                [
                    -91.137335699999994,
                    30.417353200000001
                ],
                [
                    -91.163689899999994,
                    30.417998600000001
                ]
            ]
        ]
    },
    "properties": {
        "updated": "2020-09-18T15:15:56+00:00",
        "units": "us",
        "forecastGenerator": "BaselineForecastGenerator",
        "generatedAt": "2020-09-18T15:24:45+00:00",
        "updateTime": "2020-09-18T15:15:56+00:00",
        "validTimes": "2020-09-18T09:00:00+00:00/P7DT16H",
        "elevation": {
            "value": 7.0103999999999997,
            "unitCode": "unit:m"
        },
        "periods": [
            {
                "number": 1,
                "name": "Today",
                "startTime": "2020-09-18T10:00:00-05:00",
                "endTime": "2020-09-18T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 84,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/day/ovc?size=medium",
                "shortForecast": "Cloudy",
                "detailedForecast": "Cloudy, with a high near 84. Northeast wind around 5 mph."
            },
            {
                "number": 2,
                "name": "Tonight",
                "startTime": "2020-09-18T18:00:00-05:00",
                "endTime": "2020-09-19T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 70,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/night/ovc?size=medium",
                "shortForecast": "Cloudy",
                "detailedForecast": "Cloudy, with a low around 70. Northeast wind around 5 mph."
            },
            {
                "number": 3,
                "name": "Saturday",
                "startTime": "2020-09-19T06:00:00-05:00",
                "endTime": "2020-09-19T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 83,
                "temperatureUnit": "F",
                "temperatureTrend": "falling",
                "windSpeed": "10 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/day/bkn/rain_showers,20?size=medium",
                "shortForecast": "Mostly Cloudy then Slight Chance Rain Showers",
                "detailedForecast": "A slight chance of rain showers after 1pm. Mostly cloudy. High near 83, with temperatures falling to around 81 in the afternoon. Northeast wind around 10 mph. Chance of precipitation is 20%."
            },
            {
                "number": 4,
                "name": "Saturday Night",
                "startTime": "2020-09-19T18:00:00-05:00",
                "endTime": "2020-09-20T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 70,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 to 10 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/night/rain_showers,20?size=medium",
                "shortForecast": "Slight Chance Rain Showers",
                "detailedForecast": "A slight chance of rain showers. Mostly cloudy, with a low around 70. Northeast wind 5 to 10 mph. Chance of precipitation is 20%."
            },
            {
                "number": 5,
                "name": "Sunday",
                "startTime": "2020-09-20T06:00:00-05:00",
                "endTime": "2020-09-20T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 81,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 to 15 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/day/rain_showers,30/rain_showers,40?size=medium",
                "shortForecast": "Chance Rain Showers",
                "detailedForecast": "A chance of rain showers. Cloudy, with a high near 81. East wind 10 to 15 mph. Chance of precipitation is 40%."
            },
            {
                "number": 6,
                "name": "Sunday Night",
                "startTime": "2020-09-20T18:00:00-05:00",
                "endTime": "2020-09-21T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 68,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/night/rain_showers,40?size=medium",
                "shortForecast": "Chance Rain Showers",
                "detailedForecast": "A chance of rain showers. Cloudy, with a low around 68. East wind around 10 mph. Chance of precipitation is 40%."
            },
            {
                "number": 7,
                "name": "Monday",
                "startTime": "2020-09-21T06:00:00-05:00",
                "endTime": "2020-09-21T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 79,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 to 15 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/day/rain_showers,40/tsra,40?size=medium",
                "shortForecast": "Chance Rain Showers",
                "detailedForecast": "A chance of rain showers before 1pm, then a chance of showers and thunderstorms. Cloudy, with a high near 79. East wind 10 to 15 mph. Chance of precipitation is 40%."
            },
            {
                "number": 8,
                "name": "Monday Night",
                "startTime": "2020-09-21T18:00:00-05:00",
                "endTime": "2020-09-22T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 66,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/night/tsra,40?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of showers and thunderstorms. Cloudy, with a low around 66. East wind around 10 mph. Chance of precipitation is 40%."
            },
            {
                "number": 9,
                "name": "Tuesday",
                "startTime": "2020-09-22T06:00:00-05:00",
                "endTime": "2020-09-22T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 79,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/day/rain_showers/tsra?size=medium",
                "shortForecast": "Chance Rain Showers then Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of rain showers before 1pm, then a chance of showers and thunderstorms. Mostly cloudy, with a high near 79. East wind around 10 mph."
            },
            {
                "number": 10,
                "name": "Tuesday Night",
                "startTime": "2020-09-22T18:00:00-05:00",
                "endTime": "2020-09-23T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 66,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 to 10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/night/tsra?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of showers and thunderstorms before 1am, then a chance of showers and thunderstorms. Mostly cloudy, with a low around 66. East wind 5 to 10 mph."
            },
            {
                "number": 11,
                "name": "Wednesday",
                "startTime": "2020-09-23T06:00:00-05:00",
                "endTime": "2020-09-23T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 80,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/day/tsra?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of showers and thunderstorms before 1pm, then a chance of showers and thunderstorms. Mostly cloudy, with a high near 80. East wind around 10 mph."
            },
            {
                "number": 12,
                "name": "Wednesday Night",
                "startTime": "2020-09-23T18:00:00-05:00",
                "endTime": "2020-09-24T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 67,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 to 10 mph",
                "windDirection": "E",
                "icon": "https://api.weather.gov/icons/land/night/tsra?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of showers and thunderstorms before 1am, then a chance of showers and thunderstorms. Mostly cloudy, with a low around 67. East wind 5 to 10 mph."
            },
            {
                "number": 13,
                "name": "Thursday",
                "startTime": "2020-09-24T06:00:00-05:00",
                "endTime": "2020-09-24T18:00:00-05:00",
                "isDaytime": true,
                "temperature": 81,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "10 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/day/tsra?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms",
                "detailedForecast": "A chance of showers and thunderstorms before 1pm, then a chance of showers and thunderstorms. Mostly cloudy, with a high near 81. Northeast wind around 10 mph."
            },
            {
                "number": 14,
                "name": "Thursday Night",
                "startTime": "2020-09-24T18:00:00-05:00",
                "endTime": "2020-09-25T06:00:00-05:00",
                "isDaytime": false,
                "temperature": 67,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "5 to 10 mph",
                "windDirection": "NE",
                "icon": "https://api.weather.gov/icons/land/night/tsra_sct/bkn?size=medium",
                "shortForecast": "Chance Showers And Thunderstorms then Mostly Cloudy",
                "detailedForecast": "A chance of showers and thunderstorms before 8pm. Mostly cloudy, with a low around 67. Northeast wind 5 to 10 mph."
            }
        ]
    }
}