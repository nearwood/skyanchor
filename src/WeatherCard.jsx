import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
// import Divider from '@material-ui/core/Divider';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Collapse from '@material-ui/core/Collapse';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  }
}));

export default function App(props) { //TODO prop-types
  const { period, hourlyURL } = props;

  const classes = useStyles();
  const [expanded, setExpanded] = useState(true);
  const [hourlyForecast, setHourlyForecast] = useState([]);

  useEffect(() => {
    async function fetchData() {
      const response = await fetch(hourlyURL).then(results => results.json());
      console.log(response);
      setHourlyForecast(response?.properties?.periods);
    }
    if (hourlyURL) {
      fetchData();
    }
  }, [hourlyURL]);

  return (
    <Card>
      <CardHeader
        avatar={<Avatar src={period.icon} aria-label="weather icon" />}
        title={period.name}
        subheader="September 14, 2016"
      />
      <CardContent>
        <span>{period.temperature} °{period.temperatureUnit}</span><div>{period.shortForecast}</div>
      </CardContent>
      {/* <Divider variant="middle" /> */}
      {period.number === 1 && <>
        <CardActions>
          <IconButton
            onClick={() => setExpanded(!expanded)}
            aria-expanded={expanded}
            aria-label="show hourly">
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>
        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <Table className={classes.table} size="small" aria-label="a dense table">
            <TableHead>
              <TableRow>
                <TableCell>Time</TableCell>
                <TableCell align="right">Day?</TableCell>
                <TableCell align="right">Temperature</TableCell>
                <TableCell align="right">Wind</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {hourlyForecast.map((hour) => (
                <TableRow key={hour.name}>
                  <TableCell component="th" scope="row">
                    {hour.startTime}
                  </TableCell>
                  <TableCell align="right">{hour.isDaytime ? 'day' : 'night'}</TableCell>
                  <TableCell align="right">{hour.temperature} °{hour.temperatureUnit}</TableCell>
                  <TableCell align="right">{hour.windSpeed} {hour.windDirection}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </Collapse>
      </>}
    </Card>
  );
}

/*
"properties": {
  "updated": "2020-09-17T02:16:14+00:00",
  "units": "us",
  "forecastGenerator": "HourlyForecastGenerator",
  "generatedAt": "2020-09-17T03:06:04+00:00",
  "updateTime": "2020-09-17T02:16:14+00:00",
  "validTimes": "2020-09-16T20:00:00+00:00/P8DT5H",
  "elevation": {
    "value": 7.0103999999999997,
    "unitCode": "unit:m"
  },
  "periods": [
  {
    "number": 1,
    "name": "",
    "startTime": "2020-09-16T22:00:00-05:00",
    "endTime": "2020-09-16T23:00:00-05:00",
    "isDaytime": false,
    "temperature": 79,
    "temperatureUnit": "F",
    "temperatureTrend": null,
    "windSpeed": "5 mph",
    "windDirection": "N",
    "icon": "https://api.weather.gov/icons/land/night/bkn?size=small",
    "shortForecast": "Mostly Cloudy",
    "detailedForecast": ""
  },
*/

/*
properties:
elevation: {value: 441.96, unitCode: "unit:m"}
forecastGenerator: "BaselineForecastGenerator"
generatedAt: "2020-09-15T22:00:13+00:00"
periods: Array(14)
0: {number: 1, name: "This Afternoon", startTime: "2020-09-15T17:00:00-05:00", endTime: "2020-09-15T18:00:00-05:00", isDaytime: true, …}
1: {number: 2, name: "Tonight", startTime: "2020-09-15T18:00:00-05:00", endTime: "2020-09-16T06:00:00-05:00", isDaytime: false, …}
2: {number: 3, name: "Wednesday", startTime: "2020-09-16T06:00:00-05:00", endTime: "2020-09-16T18:00:00-05:00", isDaytime: true, …}
3: {number: 4, name: "Wednesday Night", startTime: "2020-09-16T18:00:00-05:00", endTime: "2020-09-17T06:00:00-05:00", isDaytime: false, …}
4: {number: 5, name: "Thursday", startTime: "2020-09-17T06:00:00-05:00", endTime: "2020-09-17T18:00:00-05:00", isDaytime: true, …}
5: {number: 6, name: "Thursday Night", startTime: "2020-09-17T18:00:00-05:00", endTime: "2020-09-18T06:00:00-05:00", isDaytime: false, …}
6: {number: 7, name: "Friday", startTime: "2020-09-18T06:00:00-05:00", endTime: "2020-09-18T18:00:00-05:00", isDaytime: true, …}
7: {number: 8, name: "Friday Night", startTime: "2020-09-18T18:00:00-05:00", endTime: "2020-09-19T06:00:00-05:00", isDaytime: false, …}
8: {number: 9, name: "Saturday", startTime: "2020-09-19T06:00:00-05:00", endTime: "2020-09-19T18:00:00-05:00", isDaytime: true, …}
9: {number: 10, name: "Saturday Night", startTime: "2020-09-19T18:00:00-05:00", endTime: "2020-09-20T06:00:00-05:00", isDaytime: false, …}
10: {number: 11, name: "Sunday", startTime: "2020-09-20T06:00:00-05:00", endTime: "2020-09-20T18:00:00-05:00", isDaytime: true, …}
11: {number: 12, name: "Sunday Night", startTime: "2020-09-20T18:00:00-05:00", endTime: "2020-09-21T06:00:00-05:00", isDaytime: false, …}
12: {number: 13, name: "Monday", startTime: "2020-09-21T06:00:00-05:00", endTime: "2020-09-21T18:00:00-05:00", isDaytime: true, …}
13: {number: 14, name: "Monday Night", startTime: "2020-09-21T18:00:00-05:00", endTime: "2020-09-22T06:00:00-05:00", isDaytime: false, …}
length: 14
__proto__: Array(0)
units: "us"
updateTime: "2020-09-15T20:25:36+00:00"
updated: "2020-09-15T20:25:36+00:00"
validTimes: "2020-09-15T14:00:00+00:00/P7DT11H"
*/

/*
detailedForecast: "Mostly clear, with a low around 59. South wind 5 to 10 mph."
endTime: "2020-09-16T06:00:00-05:00"
icon: "https://api.weather.gov/icons/land/night/few?size=medium"
isDaytime: false
name: "Tonight"
number: 2
shortForecast: "Mostly Clear"
startTime: "2020-09-15T18:00:00-05:00"
temperature: 59
temperatureTrend: null | temperatureTrend: "falling" | temperatureTrend: "rising"
temperatureUnit: "F"
windDirection: "S"
windSpeed: "5 to 10 mph"
*/
