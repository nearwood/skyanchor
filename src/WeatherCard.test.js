import React from 'react';
import { render } from '@testing-library/react';
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import WeatherCard from './WeatherCard';

import forecast from './exampleData/forecast.json';

// test('renders learn react link', () => {
//   const { getByText } = render(<WeatherCard period={forecast.properties.periods[0]} />);
//   //const linkElement = getByText(/Today - .*/);
//   //expect(linkElement).toBeInTheDocument();
// });

let container = null;
let unmount = null;

beforeEach(() => {
});

afterEach(() => {
  unmount();
  container = null;
});

xtest("Snapshot: Example WeatherCard", () => {
  act(() => {
    const result = render(
      <WeatherCard period={forecast.properties.periods[0]} />
    );
    container = result.container;
    unmount = result.unmount;
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});