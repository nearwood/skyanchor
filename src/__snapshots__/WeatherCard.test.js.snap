// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: Example WeatherCard 1`] = `""`;
